frontend/src/metabase-lib/v1/parameters/utils/targets.ts

```diff
@@ -101,15 +101,6 @@ function getParameterTargetFieldFromFieldRef(
   }
 
   const { query, columns } = getParameterColumns(question, parameter);
-
-  if (columns.length === 0) {
-    // query and metadata are not available: 1) no data permissions 2) embedding
-    // there is no way to find the correct field so pick the first one matching by name
-    return fields.find(
```

Comment:
This is essentially how our logic for parameters<->field matching for public and embedding cases looked like before :)